src/shared/codelens/localLambdaRunner.ts

```diff
@@ -310,3 +321,235 @@ export class LocalLambdaRunner {
+        'Build complete.'
+    )
+
+    return path.join(samBuildOutputFolder, 'template.yaml')
+}
+
+export const invokeLambdaFunction = async (params: {
+    baseBuildDir: string,
+    channelLogger: ChannelLogger,
+    configuration: SettingsConfiguration,
+    debugConfig: DebugConfiguration,
+    documentUri: vscode.Uri,
+    handlerName: string,
+    isDebug?: boolean,
+    samTemplatePath: string,
+    samTaskInvoker: SamCliTaskInvoker,
+    onWillAttachDebugger?(): Promise<void>,
+}): Promise<void> => {
+    params.channelLogger.info(
+        'AWS.output.starting.sam.app.locally',
+        'Starting the SAM Application locally (see Terminal for output)'
+    )
+    params.channelLogger.logger.info(`localLambdaRunner.invokeLambdaFunction: ${JSON.stringify(
+        {
+            baseBuildDir: params.baseBuildDir,
+            configuration: params.configuration,
+            debugConfig: params.debugConfig,
+            documentUri: vscode.Uri,
+            handlerName: params.handlerName,
+            isDebug: params.isDebug,
+            samTemplatePath: params.samTemplatePath,
+        },
+        undefined,
+        2)}`
+    )
+
+    const eventPath: string = path.join(params.baseBuildDir, 'event.json')
+    const environmentVariablePath = path.join(params.baseBuildDir, 'env-vars.json')
+    const config = await getConfig({
+        handlerName: params.handlerName,
+        documentUri: params.documentUri,
+    })
+
+    await writeFile(eventPath, JSON.stringify(config.event || {}))
+    await writeFile(
+        environmentVariablePath,
+        JSON.stringify(getEnvironmentVariables(config))
+    )
+
+    const command = new SamCliLocalInvokeInvocation({
+        templateResourceName: TEMPLATE_RESOURCE_NAME,
+        templatePath: params.samTemplatePath,
+        eventPath,
+        environmentVariablePath,
+        debugPort: (params.isDebug) ? params.debugConfig.port.toString() : undefined,
+        invoker: params.samTaskInvoker
+    })
+
+    await command.execute()
+
+    if (params.isDebug) {
+        params.channelLogger.info(
+            'AWS.output.sam.local.waiting',
+            'Waiting for SAM Application to start before attaching debugger...'
+        )
+
+        const timeoutMillis = params.configuration.readSetting<number>(
+            'samcli.debug.attach.timeout.millis',
+            SAM_LOCAL_PORT_CHECK_RETRY_TIMEOUT_MILLIS_DEFAULT
+        )
+
+        await tcpPortUsed.waitUntilUsed(
+            params.debugConfig.port,
+            SAM_LOCAL_PORT_CHECK_RETRY_INTERVAL_MILLIS,
+            timeoutMillis
+        )
+
+        await attachDebugger({
+            channeLogger: params.channelLogger,
+            debugConfig: params.debugConfig,
+            outputChannel: params.channelLogger.channel,
+            /* tslint:disable:no-unbound-method */ // It's a fat arrow. No this. No binding.
+            onWillAttachDebugger: params.onWillAttachDebugger,
+})
```

Comment:
What happened to the indentation from here to the end of the file?

Reply:
Strange. I can't imagine why. Fixed.
✔